integration-tests/bats/status.bats

```diff
@@ -204,4 +204,95 @@ SQL
 @test "status: dolt reset --hard with more than one additional arg throws an error " {
     run dolt reset --hard HEAD HEAD2
     [ "$status" -eq 1 ]
+}
+
+@test "status: dolt reset hard with ~ works" {
+    dolt sql -q "CREATE TABLE test (pk int PRIMARY KEY);"
+    dolt commit -am "cm1"
+
+    dolt sql -q "INSERT INTO test values (1);"
+    dolt commit -am "cm2"
+
+    dolt sql -q "INSERT INTO test VALUES (2);"
+    dolt commit -am "cm3"
+
+    # Do a hard reset back one commit and confirm the appropriate values.
+    run dolt reset --hard HEAD~1
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT sum(pk) FROM test;"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "1" ]] || false
+
+    # Since this is a hard reset double check the status
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+
+    # Run again with ~2 this time
+    dolt sql -q "INSERT INTO test VALUES (2);"
+    dolt commit -am "cm3"
+
+    # Do a hard reset back two commits and confirm the appropriate values.
+    run dolt reset --hard HEAD~2
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "SELECT sum(pk) FROM test;"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "NULL" ]] || false
+
+    # Since this is a hard reset double check the status
+    run dolt status
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "On branch master" ]] || false
+    [[ "$output" =~ "nothing to commit, working tree clean" ]] || false
+}
+
+@test "status: dolt reset soft with ~ works" {
```

Comment:
You have logic in the command about updating staged roots, but you aren't testing any of that logic here. Add some plz